network/p2p/p2p.go

```diff
@@ -260,57 +273,99 @@ func (n *p2pNetwork) UpdateSubnets(logger *zap.Logger) {
 	for ; true; <-ticker.C {
 		start := time.Now()
 
-		// Compute the new subnets according to the active validators.
-		newSubnets := make([]byte, commons.Subnets())
-		copy(newSubnets, n.subnets)
-		n.activeValidators.Range(func(pkHex string, status validatorStatus) bool {
-			// TODO: alan - fork support
-			pbBytes, err := hex.DecodeString(pkHex)
-			if err != nil {
-				return false
-			}
-			// TODO: alan - optimization to get active committees only
-			share := n.nodeStorage.Shares().Get(nil, pbBytes)
-			if share == nil {
-				return false
-			}
-			cid := share.CommitteeID()
-			subnet := commons.CommitteeSubnet(cid)
-			newSubnets[subnet] = byte(1)
+		// Compute the new subnets according to the active committees/validators.
+		updatedSubnets := make([]byte, commons.Subnets())
+		copy(updatedSubnets, n.fixedSubnets)
+
+		n.activeCommittees.Range(func(cid string, status validatorStatus) bool {
+			subnet := commons.CommitteeSubnet(spectypes.CommitteeID([]byte(cid)))
+			updatedSubnets[subnet] = byte(1)
 			return true
 		})
-		n.subnets = newSubnets
+
+		if !n.cfg.Network.PastAlanFork() {
+			n.activeValidators.Range(func(pkHex string, status validatorStatus) bool {
+				// TODO: alan - fork support
+				pbBytes, err := hex.DecodeString(pkHex)
+				if err != nil {
+					return false
+				}
+				// TODO: alan - optimization to get active committees only
+				share := n.nodeStorage.Shares().Get(nil, pbBytes)
+				if share == nil {
+					return false
+				}
+				cid := share.CommitteeID()
+				subnet := commons.CommitteeSubnet(cid)
+				updatedSubnets[subnet] = byte(1)
+				return true
+			})
```

Comment:
Isn't this implementing the alan logic
```go
subnet := commons.CommitteeSubnet(cid)
```
while inside "genesis if" (`!n.cfg.Network.PastAlanFork()`) ?

Reply:
you're right, we should take this part out of the if and only include in the if the pre-fork behavior

Reply:
should look lie this:

```go
		n.activeValidators.Range(func(pkHex string, status validatorStatus) bool {
			subnet := commons.ValidatorSubnet(pkHex)
			updatedSubnets[subnet] = byte(1)
			return true
		})

```

Reply:
Okk, updating